src/coreclr/jit/morph.cpp

```diff
@@ -3728,7 +3729,19 @@ GenTree* Compiler::fgMorphMultiregStructArg(CallArg* arg)
 #elif defined(TARGET_LOONGARCH64)
             if (arg->AbiInfo.StructFloatFieldType[inx] != TYP_UNDEF)
             {
-                elems[inx].Type = arg->AbiInfo.StructFloatFieldType[inx];
+                var_types Type = arg->AbiInfo.StructFloatFieldType[inx];
+                if (TYP_I_IMPL == Type)
+                {
+                    Type = getSlotType(inx);
+                    Type = varTypeIsGC(Type) ? Type : TYP_I_IMPL;
+
+                    elems[inx].Type                        = Type;
+                    arg->AbiInfo.StructFloatFieldType[inx] = Type;
```

Comment:
Mutating `AbiInfo` outside of `AddFinalArgsAndDetermineABIInfo` is not an acceptable solution.

Instead of this:

1) In `genCall` you can use `argNode->AsPutArgSplit()->GetRegTypeByIndex(0)`.
2) Here you can do the same thing Unix x64 code above does:
```cpp
if (!varTypeIsGC(getSlotType(inx)) && (arg->AbiInfo.StructFloatFieldType[inx] != TYP_UNDEF))
{
    elems[inx].Type = arg->AbiInfo.StructFloatFieldType[inx];
    offset += (structSize > TARGET_POINTER_SIZE) ? 8 : 4;
}
```

Reply:
> Mutating `AbiInfo` outside of `AddFinalArgsAndDetermineABIInfo` is not an acceptable solution.

yes, it's better to do it within the `AddFinalArgsAndDetermineABIInfo` for uniformity but here will save an extra getting the gctypes within the `AddFinalArgsAndDetermineABIInfo`.

Reply:
Yeah, that's why I suggested leaving `AddFinalArgsAndDetermineABIInfo` be for now and get the GC types in multi-reg morphing instead.

However, that may not actually be too necessary, as getting the GC types in `AddFinalArgsAndDetermineABIInfo` should be pretty cheap as well -- using the struct layout from the node, which we query anyways to get the size.

Reply:
Thanks, I had upload patch.